adds default alt values to inline images and the image block

antpb opened the pull request:
Sets default alt values to fix #1520 
Previous conversation can be found here: https://github.com/WordPress/gutenberg/pull/10482

How to test:
create image block
add image with no alt value
inspect image to see that alt value is now set as a readable string "This image has an empty alt attribute, its file name is "FILENAME""

chrisvanpatten committed:
Deprecate PanelColor components (#10391)

* Deprecate PanelColor components

* Update deprecation version numbers

* Fix a test broken during the rebase

daniloercoli committed:
Refactor `onSplit` and `insertBlockAfter` in RichText component, Para and Heading blocks for mobile, tryin gto stay closer to the web version. (#10690)

aduth committed:
Blocks: Add direction attribute to paragraph block (#10663)

Soean committed:
bold permalink label (#10566)

talldan committed:
Fix autocomplete keyboard nav in link popover (#10716)

* Fix autocomplete cursor key navigation

* Avoid running onClickOutside logic for other non-onClickOutside events

* Add e2e tests for keyboard interaction with link container

danielbachhuber committed:
Update REST Search controller to use 'wp/v2' namespace (#10670)

* Update REST Search controller to use 'wp/v2' namespace

* Docs: Update REST API url

jasmussen committed:
Polish tertiary actions, switches, save state (#10552)

* Polish Save Draft/Switch to Draft area
* Use a Button for Trash button
* Polish the toggles.

jorgefilipecosta committed:
Fix: Font size default classes. (#10719)

jasmussen committed:
Columns Block: Make responsive, make editor and theme match (#10541)

* Add basic responsiveness.

* Refactor columns metrics. Level the playing field in editor and frontend.

* Add space between colums.

Fixes #7818.
Fixes #6048.

* Remove "beta" designation from columns block.

* Columns block: Fix column width when editing

* Column block: Improve padding for the first and last item in a row

* Tests: Rename Columns block name also in e2e tests

youknowriad committed:
Remove 4.1 deprecated features (#10732)

jorgefilipecosta committed:
Rename cover image to cover; Add video in cover block; (#10659)

## Description
This commit renames cover image block into cover and adds video support o cover.
The approach used to rename the block is the same used for the move of core/text to core/paragraph.
A change in api/parser.js. This approach may be seen as a temporary solution until a final approach to this use case exists, and we feel comfortable in exposing an API for this.

Transformations were updated, and a video transform was added.

## How has this been tested?
Verify the cover block still works as expected, containing the same functionality the existed in the cover image.
Verify it is possible to set video as background in the cover block.

daniloercoli committed:
Handle the case when the split occurs at the trailing or leading edge of the field. (#10737)

aduth committed:
Writing Flow: Avoid horizontal writing flow navigation in native inputs (#9978)

talldan committed:
Add 'insert from url' for image block (#9264)

* Extract isBlobURL into blob package and update usage

Co-authored-by: imath <math.viet@gmail.com>

* Add ability to insert an image in image block using a url

Co-authored-by: imath <math.viet@gmail.com>

* Position url input underneath upload file buttons and apply margin

Co-authored-by: imath <math.viet@gmail.com>

* Refactor styles to avoid use of element selector

Co-authored-by: imath <math.viet@gmail.com>

* Ensure form elements have bottom margin to separate them on tiny screens

Co-authored-by: imath <math.viet@gmail.com>

* Change type of block used in e2e test to a separator

As mentioned in the comment above, this test will not work for blocks that
contain an input. The image block now contains a url input field causing
the test to fail.

Co-authored-by: imath <math.viet@gmail.com>

* Fix invalid reference to this.isBlobURL

* Add expandable section for url form on media placeholder

* Change icon for media placeholder button to left/right chevron

* Update snapshots

* Try: expanding url input based on whether text is entered

* Try: use link popver for Insert from URL in media placeholder

* Implement URLPopover in MediaPlaceholder

* Use consistent declaration of defaults for props

* Use a generalized selector for buttons in the media placeholder

* Improve copy in URL input placeholder

* Make behaviour when selecting a differnent url the same as other media blocks

* Update snapshot tests

* Revert "Change type of block used in e2e test to a separator"

This reverts commit 6c8546c29070827d62d37b1f0198ad302f79ec49.

* Use onClose over onClickOutside (onClose implements onClickOutside)

* Rename functions

* Address code review feedback

* Update changelog for blob package

* Updated changelog for editor package

noisysocks committed:
Add Custom Fields and Advanced Panels to Options modal (#10676)

* First pass at adding 'Advanced Panels' to Options modal

Allows the user to enable or disable 'Advanced Panels' (AKA Meta Boxes)
via the Options modal.

chrisvanpatten committed:
Fix a Text Selection Exception in Safari (#8255)

* Catch a Text Selection Exception in Safari; fixes #8254

* Improve the inline explanation of why we need a try/catch block

notnownikki committed:
Make responsive embeds a theme option (#10477)

pento committed:
chore(release): publish

 - @wordpress/blob@2.0.3
 - @wordpress/block-library@2.1.2
 - @wordpress/blocks@4.0.3
 - @wordpress/components@4.1.2
 - @wordpress/compose@2.0.4
 - @wordpress/core-data@2.0.4
 - @wordpress/data@2.1.3
 - @wordpress/dom@2.0.3
 - @wordpress/edit-post@1.0.2
 - @wordpress/editor@4.0.3
 - @wordpress/element@2.1.3
 - @wordpress/escape-html@1.0.0
 - @wordpress/list-reusable-blocks@1.1.2
 - @wordpress/nux@2.0.4
 - @wordpress/plugins@2.0.4
 - @wordpress/redux-routine@3.0.2
 - @wordpress/rich-text@1.0.0
 - @wordpress/viewport@2.0.4

blowery committed:
Update clipboard dependency to at least 2.0.1 (#10739)

* Update clipboard dependency to at least 2.0.1

This upgrades the clipboard dependency to at least 2.0.1, which resolves a possible conflict with the browser global Clipboard.

See https://github.com/zenorocha/clipboard.js/releases for release notes

* take the update to package-lock offered by npm i

* take another update to package-lock from a second run

* package-lock after removing lock entirely and rerunning npm i

* Update package-lock.json

* Update package-lock.json

* Update package-lock.json

notnownikki committed:
Make sure autocomplete triggers are regex safe (#10785)

youknowriad committed:
Replace gutenberg in classNames with block-editor (#10783)

notnownikki committed:
Fix WordPress block resolution and embeds as reusable blocks (#10035)

* Fix WordPress block resolution and embeds as reusable blocks

* Let's not infinitely recurse and crash.

* Reduce the amount of return points

* Doc update

Commit:
Docs: Update custom icon link (#10784)

Provided a better link for information about adding a custom icon. Linking to the wp Element page doesn't provide any specific information about adding a custom icon whereas the block api page does.

mcsf committed:
Try: Ensure compatibility with WordPress 5.0 (#10754)

* Don't load conflicting PHP entities

* Systematically override core scripts

* Systematically override core styles

jorgefilipecosta committed:
Add "Text & Image" Block (#9416)

This commit implements a block that contains two areas one for a media element (image/video) and one area for blocks (buttons, paragraphs etc). The elements of both areas are vertically aligned.
Here instead of middle blocks, the parent block will self-contain the media area (that allows images or videos), so we have only one simple nested area for the content.

notnownikki committed:
Use mock response for demo test if Vimeo is down (#10720)

ryelle committed:
ColorPalette: Add an accessible color picker based on ChromePicker (#10564)

* Add color picker component

* Add color picker styles to build

* Update to buttons for focus handles

* Update styles

* Update class name

* Add color-picker component to exports

* Update name to ColorPicker

* Add labels to color input groups

* Fix describedby prop name

* Update view to use `getDerivedStateFromProps`

* Add “Input” component to handle intermediate state when entering values

Specifically needed for hex colors, which can be invalid while they’re being entered

* Add space around text inputs

* Translate toggle label

* Add focus styles to colourpicker pointers.

* Remove react-color dependency

* Add class name to better find the component in the devtools

* Add tinycolor2 dependency

* Add reference to react-color license & copyright

* Better label

* Update snapshot

* Decouple valueKey and label

so we can use semantic labels.

* Use KeyboardShortcuts component to avoid separate instances of Mousetrap

* Use createRef

* Normalize HSV / HSL values

H: 0 - 260
S, L, V: 0 - 100

* Omit `valueKey` from being passed to input

* Use createRef in all components

* Update saturation handle label

* Prevent key events on slider handles

Fixes the issue where VoiceOver grabs arrow-key input as navigation

* Use a focusable div for slider handle

* Apply focus style to alpha handle

* Add description to hue controls

* Add speak commands for view change

* Update snapshot

* Remove test as the component API surface changed

We have done some changes to the ChromePicker so this test
can no longer be replicated.

* Add README, tests and update Changelog

* Remove react-color dependency

* Add docs manifest file

brandonpayton committed:
Address review of unregistered type handler (#10776)

ryelle committed:
Color Picker: Fall back to min-width of popover (#10805)

afercia committed:
Make the color picker saturation control work with windows screen readers. (#10807)

aduth committed:
Components: Add `info` prop support to MenuItem, FeatureToggle (#10802)

* Components: Allow aria-label override on IconButton

* Components: Add `info` prop support to MenuItem

* Edit Post: Add `info` prop support to FeatureToggle

* Edit Post: Add feature toggle info texts

* Components: Menu Item: Break text beyond mobile toolbar

* Components: Icon Button: Indent text by SVG margin

Avoids offset indent on wrapped text

* Tweak menu item styling

* Avoid italic text

aduth committed:
REST API: Use posts endpoint for reusable blocks (#10751)

* REST API: Use posts endpoint for reusable blocks

* REST: Restore WP_REST_Blocks_Controller for permissions check

* Reusable Blocks: Enable post listing edit

* Reusable Blocks: Trash on delete action

* List Reusable Blocks: Provide context to post request

* Reusable Blocks: Verify edit capability on export action

* List Reusable Blocks: Import as published

ArnaudBan committed:
Docs: Fix deprecated-blocks example (#10717)

## Description
I changed a small error in the deprecated-blocks documentation

## How has this been tested?
I looked in my Github repo if the markdown file preview was ok

## Types of changes
Typo in documentation

mtias committed:
Update plugin version to 4.1 RC1. (#10749)

tofumatt committed:
Oops, forgot to update these to master

youknowriad committed:
Update the published packages changelog + revert lerna changelog config (#10819)

youknowriad committed:
chore(release): publish

 - @wordpress/blob@2.1.0
 - @wordpress/block-library@2.1.3
 - @wordpress/blocks@4.0.4
 - @wordpress/components@4.2.0
 - @wordpress/compose@2.0.5
 - @wordpress/core-data@2.0.5
 - @wordpress/data@2.1.4
 - @wordpress/dom@2.0.4
 - @wordpress/edit-post@1.0.3
 - @wordpress/editor@5.0.0
 - @wordpress/element@2.1.4
 - @wordpress/escape-html@1.0.1
 - @wordpress/list-reusable-blocks@1.1.3
 - @wordpress/nux@2.0.5
 - @wordpress/plugins@2.0.5
 - @wordpress/redux-routine@3.0.3
 - @wordpress/rich-text@1.0.1
 - @wordpress/viewport@2.0.5

ntwb committed:
Revert 123811df9613cdf9dae3e9ec8372166b1b3a42e4 (#10823)

youknowriad committed:
Fix importing react-dates styles into the components package stylesheet (#10831)

youknowriad committed:
Fix fullscreen mode (#10830)

* Fix fullscreen mode

* Add e2e test for fullscreen mode

youknowriad committed:
Update changelog for the last packages release (#10836)

mcsf committed:
Rename PHP functions to prevent conflict w/ core (#10826)

These functions (e.g. `get_block_categories`) are declared in the plugin
before their counterparts in core, which means that guarding the
plugin's function declarations with

	if ( ! function_exists( 'foo' ) ) {

doesn't make a difference, and the collisions will result in server
fatals.

Presskopp committed:
Update rest-api.php (#10840)

add missing 'to' in string "not allowed _ make"

audrasjb committed:
Remove closing </li> tag from i18n string (#10849)

ocean90 committed:
Avoid PHP notices due to non-available meta boxes (#10850)

afercia committed:
Improve the Toggle Control elements DOM order for better accessibility. (#10870)

afercia committed:
Set correct media type for video poster image and manage focus. (#10864)

kadamwhite committed:
Implement fetchAllMiddleware to handle per_page=-1 through pagination (#10762)

youknowriad committed:
Update plugin version to 4.1 RC2. (#10893)

gziolo committed:
chore(release): publish

 - @wordpress/api-fetch@2.1.0
 - @wordpress/block-library@2.1.4
 - @wordpress/components@4.2.1
 - @wordpress/core-data@2.0.6
 - @wordpress/edit-post@1.0.4
 - @wordpress/editor@5.0.1
 - @wordpress/list-reusable-blocks@1.1.4
 - @wordpress/nux@2.0.6

pento committed:
Update Package Changelogs (#10887)

* Update package changelogs

* Bump the changelog date, because it's now Monday for everybody.

aduth committed:
Components: Avoid SlotFill props destructuring (#10921)

swissspidy committed:
Add some new url helper functions (#10885)

* Add some new url helper functions

See #10879.

* Document new functions in readme

* Improve docs

* Add another test for removeQueryArgs()

daniloercoli committed:
Remove the root tags returned by Aztec by using the correct tag reference (#10797)

blowery committed:
Update react-click-outside to 3.0 (#10748)

* Update react-click-outside to 3.0

* update package-lock

jorgefilipecosta committed:
Fix:  Show resizer on "Media & Text" block on unified toolbar mode (#10913)

## Description
Fixes: https://github.com/WordPress/gutenberg/issues/10895
On unified toolbar mode, it was not possible to resize the media on Media & Text block.
During the elaboration of https://github.com/WordPress/gutenberg/pull/9416 this bug was addressed but I think during subsequent rebases and changes more concretely the usage of our abstracted Resizable box we had a regression.

This PR applies exactly the same fix that was applied in spacer block to the Media & Text block.


## How has this been tested?
I checked it is possible to resize the media in Media & Text block in all modes including the unified toolbar.

jasmussen committed:
Try WordPress logo animation for preview (#10896)

* Try WordPress logo animation for preview

This adds a WordPress logo that paints itself when you click preview.

Labelled "Try" because it still needs a few thoughts. We might need to speed up the animation because most previews are super fast and you barely see the animation.

* Clean up CSS, add vendor prefixes

* Tweak the changelog

ellatrix committed:
Fix RichText shortcuts (#10901)

jasmussen committed:
Fix regression with mobile toolbar (#10936)

* Fix regression with mobile toolbar

This PR fixes two issues:

1. The block traversal button is hidden, it doesn't fit with translations.
2. Pinned extensions are hidden, they don't fit or even scale.

To expand on 1, we have in place a "clickthrough" method for blocks with nested contexts. One tap, for example, on a fully populated columns block selects the columns block. Another tap selects a column. A third tap, selects paragraphs inside, etc. To this argument, I think it's fair to hide the traversal button.

To expand on point 2, this was always the intent, and it's the reason every extension has to have a menu option also.

* Update changelog.

* Fix changelog.

notnownikki committed:
Embed block code reorg (#10793)

jasmussen committed:
Make options modal mobile friendly. (#10944)

This PR makes the options modal fit on mobile. Fixes #10894.

Soean committed:
Datepicker: Consistent help text (#10922)

mmtr committed:
DateTimePicker component: Use color function for defining the background (#10943)

diegoreymendez committed:
RNMobile - Force native update when splitting blocks. (#10897)

* Handle the case when the split occurs at the trailing or leading edge of the field.

* Forces to update the native components on split.

adamsilverstein committed:
Speak: Add/complete audible feedback for link editing (#10795)

* Speak ‘link removed’ when a link is removed

* Speak ‘link edited’ when a link is edited

Soean committed:
remove keyword (#10949)

diegoreymendez committed:
RNMobile - Simplifies onEnter handling. (#10902)

* Handle the case when the split occurs at the trailing or leading edge of the field.

* Forces to update the native components on split.

* Simplifies onEnter logic.

mcsf committed:
Clean-up: rm unused variable fallbacks in RichText (#10688)

chrisvanpatten committed:
Add "img" as a keyword for the image block (#10955)

* Add "img" as a keyword for the image block

* Add comment for future maintainers on why we don't translate "img"

jasmussen committed:
Fix changelog issue. (#10953)

* Fix changelog issue.

This PR fixes a typo in a previous changelog change, which happened because the 5.0.1 version was already published. This bumps it to 5.0.2.

* Address feedback.

* Address feedback?

Luciaisacomputer committed:
Fix: Inserter Search Results Not Announcing  (#10755)

* added functionality to get the result count and announcer it in the live region

* chore: Tweak variable names/readability

* Sort props

antpb committed:
adds default alt values to inline images and the image block

antpb committed:
adds default alt values to inline images and the image block

Copons committed:
File Block: Set the download attribute as empty (#10948)

georgeh committed:
Only silence REST errors if the REST server is added (#10903)

* Only silence REST errors if the REST server is added

If the REST server is not available in a code run, Gutenberg will fatal with `Fatal error: Uncaught Error: Call to undefined function gutenberg_silence_rest_errors()`

`gutenberg_silence_rest_errors` is defined in `lib/rest-api.php`, which is only loaded if `class_exists( 'WP_REST_Controller' )` in `load.php`.

This PR adds the same `class_exists()` test to the call to `gutenberg_silence_rest_errors()` in `gutenberg.php`.

swissspidy committed:
Add has-dates class to Latest Posts block if applicable (#10727)

Fixes #10691.

swissspidy committed:
Add user locale api-fetch middleware (#10862)

* Add user locale api-fetch middleware

See #10811

* Don't modify existing _locale query arg

* Add changelog entry

* Update changelog

* Use new hasQueryArg helper function

See #10885.

* Match on simply the route path, instead of malformed query param

* Fully mock embedding test

* Skip the embedding test until we can make the travis env mock the requests

* Skip demo test too, for now

Commit:
Introduce link target in image block (#9520)

jasmussen committed:
Rephrase responsive toggle. (#10981)

This replaces #10424. That rebase didn't work for me.

mcsf committed:
Autocomplete: Support non-Latin inputs (#10961)

In JavaScript regular expressions, `\w` and `\W` only match ASCII-based
characters; for example, "a" to "z", "A" to "Z", "0" to "9" and "_".
`\w` was used presumably to weed out any whitespace-containing
autocompleter inputs. Instead, look for non-whitespace characters.

talldan committed:
Ensure buttons on end of row in media-placeholder have no margin-right (#10977)

* Ensure buttons on end of row in media-placeholder have no margin-right

* Use the grid-size-small variable instead of a random 5px size

Copons committed:
Add Core styles to the SelectControl component (#10957)

youknowriad committed:
Expose setUnregisteredTypeHandlerName/getUnregisteredTypeHandlerName for mobile (#10991)

chrisvanpatten committed:
Add space above code editor toolbar (#10898)

jasmussen committed:
Fix icon size regression. (#10987)

* Fix icon size regression.

This is an alternative to #10938 which for whatever reason I can't rebase. The purpose is the same:

This PR fixes an icon size regression that was introduced in #9828. Basically SVG icons are _between 20 and 24px_.

If an SVG has an explicit width, it uses that width, within the boundaries of those extremes. But unfortunately it also means if the SVG _doesn't_ have an explicit width or height, it collapses to the min-width and min-height.

This PR updates documentation, and adds explicit dimensions for all block icons used.

Before:

![screen shot 2018-10-23 at 11 34 25](https://user-images.githubusercontent.com/1204802/47352448-32d6d000-d6ba-11e8-8ab8-c83fea8e8b07.png)

After:

![screen shot 2018-10-23 at 11 39 07](https://user-images.githubusercontent.com/1204802/47352456-35392a00-d6ba-11e8-832e-a45f3772389c.png)

* Pretty up the code a bit.

Luciaisacomputer committed:
Fix/categories block label missing (#10912)

* added withInstanceId HoC to pass in unique instance. Created a label element for the dropdown element using the unique instance ID. Applied Categories in the i18n helper for the label. Refactored a few props references to clean up the code

* switch to id instead of name

* statically set class name in constant

* Move lodash to external dependencies section

* Change the name of the select id

pento committed:
Don't set a value on the file block download attribute (#10976)

* Set the download attribute without a value

* Integration test typoes

aduth committed:
Hooks: Remove redundant hooks initialization (#10950)

notnownikki committed:
Fix embedding and demo tests (#10982)

notnownikki committed:
Do not make WordPress embeds responsive (block opt-out of responsiveness) (#10985)

ajitbohra committed:
Docs: Update link text for manifest (#10804)

daniloercoli committed:
Wire onBackSpace in RichText component (#10997)

sarahmonster committed:
Implement more generic pre-publish prompts; fixes #10750. (#11009)

notnownikki committed:
Add documentation for responsive-embeds theme option (#11011)

* Add documentation for responsive-embeds theme option

* Tweak the docs

brandonpayton committed:
Fix multi-selected warning block highlight (#10818)

ocean90 commented on packages/editor/src/components/rich-text/core-tokens/image/index.js:
```diff
@@ -35,7 +35,11 @@ export const settings = {
-	save( { id, url, alt, width } ) {
+	save( { id, url, alt, width, filename } ) {
+		if ( ! alt ) {
+			alt = sprintf( __( 'This image has an empty alt attribute; its file name is "%s"' ), filename );
```
The issue #1520 is only about adding this text to images in editing context. The `save()` method is what gets saved into the database and is visible on the front end. So this change seems wrong?

antpb replied:
I can change this to only be the edit context. I set the alt if it is empty on the save method because from my understanding, screen readers will try and read the full image url if there is no alt attribute defined.

ocean90 replied:
Related chat on Slack where this has been discussed: https://wordpress.slack.com/archives/C02SX62S6/p1540406745000100

tldr;: empty alt texts are fine, suggested change should only apply in editing context.

schlessera committed:
Add missing `@return` tag to `gutenberg_meta_box_save_redirect()` function (#11001)

The `@return` tag is missing in the docblock of the `gutenberg_meta_box_save_redirect()` function.

The function returns the modified location.

Commit:
Docs: Fix example in creating-dynamic-blocks.md (#10786)

* Update creating-dynamic-blocks.md

While posts are loading inside the edit method the `posts` variable is null, attempting to access `post[0]` causes an error 

"TypeError: Cannot read property '0' of null" and the block displays "This block has encountered an error and cannot be previewed." 

checking that posts has a value resolves this.

* Added spacing to improve readability

jasmussen committed:
Try: Fix Safari issue where hover outlines sometimes lingered (#10892)

Hopefully fixes #10741.

Safari could leave some hover outline "residue" when fading out the outlines of a block that got deselected. This PR appears to fix this.

I note "appears", because I would like other Safari users to verify this is the case.

The `will-change: transform;` property essentially says: paint this animation using the GPU, which very often solves animation related issues like this. But it sometimes causes glitches elsewhere, notably around fixed positioning, overflow, that sort of thing. In my testing, everything is fine, but please verify.

youknowriad committed:
Update plugin version to 4.1.0 (#11022)

Soean committed:
override overflow hidden (#10517)

brandonpayton committed:
Clarify selected sidebar tab for a11y (#10927)

jasmussen committed:
Small followup fix to options modal tweaks (#10978)

In https://github.com/WordPress/gutenberg/pull/10944 the options modal was made mobile friendly. This PR follows up and in fact removes the min width rule because it was redundant.

nosolosw committed:
Include viewport size in logic to show publish toggle or button (#10941)

jasmussen committed:
Fix issue with tertiary button hit areas (#10888)

* Fix issue with tertiary button hit areas

This addressed feedback in https://github.com/WordPress/gutenberg/pull/10552#issuecomment-431670148.

It makes the tertiary button style be born with the same hit area as a button, because it is intended to be used as such, just with a different styling.

* Address feedback.

* Clarify comment.

jasmussen committed:
Fix issue with unified toolbar not fitting (#10990)

* Fix issue with unified toolbar not fitting

Fixes #10959.

This PR fixes an issue where at certain viewport widths, there wouldn't be room for the unified toolbar between the icons on the left, and the buttons on the right, in the editor bar. It does so by:

- Reducing the padding between toolbar segments
- Introducing a new breakpoint, xlarge (1080px), that allows the toolbar to stay in the editor bar for longer, before it stacks.

* Revert to fullheight separators.

swissspidy committed:
Make preview placeholder text translatable (#10880)

* Make preview placeholder text translatable

* Escape i18n strings since they are directly added to markup

* Use renderToString() and decodeEntities()

* Address feedback

ocean90 committed:
Add key event handler to activate block styles with keyboard (#10909)

nosolosw committed:
Fix schedule and then publish flow (#11013)

* Use props instead of derived state to render the component.

There was a bug that happened when:

- a post was scheduled for a future date from now
- then the data changed to a past date from now

At this point, the post couldn't be published directly
by clicking the "Publish" button. This component
would show the post-publish panel with a "Post scheduled" status.

* Test: check that pre-publish panel is shown

* Test: check that post-publish panel is shown if post is published

* Test: check that post-publish panel is shown if post is scheduled

* Test: isScheduled is true, but isBeingScheduled false

Although the post status is 'future', the data is before now,
so we should show the pre-publish panel with the Publish button.

* Fix isSaving logic

* Update snapshots

* Test: check that spinner is shown if post is being saved

* Remove unused prop

* Name spinner component so snapshot test is more semantic

* Fix typo

* Update snapshot

ocean90 committed:
Replace placeholder handling for MediaPlaceholder instructions (#10861)

Add default title and instructions labels to MediaPlaceholder based on allowed types.

gziolo committed:
Testing: Add missing e2e tests for Plugins API (#10513)

* Testing: Add missing e2e tests for Plugins API

* Make all plugin API e2e test independent

youknowriad committed:
Add an eslint rule to use cross-environment SVG primitives (#10652)

* Add an eslint rule to use cross-environment SVG primitives

* Extend the list of forbidden elements
Uses an existing react eslint rule to perform validation.
Adds new components to make the existing icons mobile app friendly.

* Change the way the default icon for block is set

* Ensure that default icon goes through the process of width and height normalization in BlockIcon
Props to @jasmussen for helping to catch it.

johngodley committed:
Remove cover image ‘strong’ style (#11036)

It overrides the default so the bold doesn’t appear

gziolo committed:
Fix linter errors in master branch! (#11051)

Soean committed:
Wordcount: Return 0 if text is empty (#10602)

* Return 0 if text is empty

* return 0 with empty stirng

* return 0 for enpty string

* small typo

* Add message to changelog

afercia committed:
Improve the style variation control aria-label (#11045)

* Improve the style variation control aria-label.

* Further simplify the aria-label to just use the style label or name.

sarahmonster committed:
Use better help text for ALT text input; fixes #8391. (#11052)

jasmussen committed:
Improve handling of centered 1-col galleries with small images (#11040)

* Improve handling of centered 1-col galleries with small images

If you insert a gallery that has a single column, and multiple smallish images, then center it, the images inside do not appear centered.

This is technically correct, since the _gallery itself_ is centered, but because it's full width and because the images might vary in widths, this doesn't appear to make any difference to the content inside. So although "correct", it's not a good user experience.

This PR simply centers the content also.

* Remove editor specific alignments.

ajitbohra committed:
Date: Mark getSettings as experimental (#10636)

* Date: Mark getSettings as experimental

* Update all references to __experimentalGetSettings

* Bump deprecated version

* Add change to changelog

* Remove deprecated from 4.2

Co-Authored-By: ajitbohra <ajit@lubus.in>

* Marks as unreleased

Co-Authored-By: ajitbohra <ajit@lubus.in>

webmandesign committed:
Media & Text: Fixing vertical alignment of the image (#11025)

youknowriad committed:
Fix dynamic blocks not rendering in the frontend (#11050)

* Fix dynamic blocks not rendering in the frontend

* Use variables for hook priorities

* Test that dynamic blocks are rendered when meta boxes use the excerpt

* Fix autop hook order

* Fix linting

* Fix scoping issues

* Add an e2e test for the autop issue

* copy/paste issues

* Trying to fix travis

* Trim content to avoid small differences between themes

antpb committed:
fixing

antpb committed:
fixing

schlessera committed:
Improve preloading request code (#11007)

This removes the conditional branching based on the existence of the `get_compact_response_links()` method on the `$server` object, as that method was introduced with WordPress 4.5, and Gutenberg requires WordPress 4.9.8 or later.

Also, it turns the indirect and slow call through `call_user_func()` into a direct method call.

Note: The call would actually be more correct as a static call (`$server::get_compact_response_links( $response)`), but this notation is PHP 5.3+ only.

mcsf committed:
Update plugin version to 4.1.1 (#11078)

schlessera committed:
Fix usage of `preg_quote()` (#10998)

The end tag pattern is currently generated like this:
```
$end_tag_pattern = '/<!--\s+\/wp:' . str_replace( '/', '\/', preg_quote( $block_name ) ) . '\s+-->/';
```

Instead of separately using a string replace to replace `/` with `\/`, it is preferable to use the second argument `$delimiter` that the `preg_quote()` function provides.

See http://php.net/manual/en/function.preg-quote.php

antpb committed:
fixes regex to include file extension

antpb committed:
fixes regex to include file extension

antpb committed:
cleans up saved default alt values

antpb committed:
cleans up saved default alt values

antpb committed:
Merge branch 'master' into update-adds-default-alt-values-for-#1520

antpb commented:
@ocean90 Thanks for the catch. :) I've updated the PR to only apply the default alt text in the editing context if no `alt` is defined. I also tested adding alt text to the inspector settings and it updates from the default to the new inspector alt value as expected.

If you have any recommendations on how to better optimize this approach, I'd really appreciate it.

antpb committed:
restoring the old syntax

antpb committed:
fixes merge conflicts

aduth commented on packages/block-library/src/image/edit.js:
```diff
@@ -512,6 +541,45 @@ class ImageEdit extends Component {
 							}
 							/* eslint-enable no-lonely-if */
+							if ( ! alt ) {
+								return (
```
There's a ton of code duplication here. What's actually different? Can we refactor to limit the variation of this specific condition? i.e. if it's to return `imgNoAlt` instead of `img`, just change that in the existing return value:

```
{ alt ? img : imgNoAlt }
```

aduth commented on packages/block-library/src/image/edit.js:
```diff
@@ -58,7 +58,9 @@ const LINK_DESTINATION_CUSTOM = 'custom';
 export const pickRelevantMediaFiles = ( image ) => {
-	return pick( image, [ 'alt', 'id', 'link', 'url', 'caption' ] );
+	return {
+		...pick( image, [ 'alt', 'id', 'link', 'url', 'caption' ] ),
```
Why spread.

antpb replied:
I actually reverted this change prior to the nasty rebase attempt. The spread made sense in a prior iteration where I was changing the `alt` value. Thanks for calling out. :)

aduth commented:
You're going to have an easier time against merge conflicts if you rebase and keep to a single commit.

https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase

antpb commented:
:( I made this PR really messy with that last rebase attempt. Sorry.  I've added a new, more final PR that should be easier to review/merge: https://github.com/WordPress/gutenberg/pull/11218
